src/main/scala/com.snowplowanalytics.dataflow/streaming/SimpleEvent.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2017 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.dataflow.streaming
+
+// Java
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.DateTime
+
+// json4s
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+import org.json4s.ext.JodaTimeSerializers.all
+
+// This project
+import storage.BucketingStrategy
+
+//Scala
+import scala.util.Try
+
+/**
+ * Companion object for creating a SimpleEvent
+ * from incoming JSON
+ */
+object SimpleEvent {
+
+  //Singleton object to be able to parse our DateTime format
+  case object CustomDateTimeSerializer extends CustomSerializer[DateTime](format => (
+    {
+      case JString(s) => new DateTime(s)
+      case JNull => null
+    },
+    {
+      case d: DateTime => JString(format.dateFormat.format(d.toDate))
+    }
+  ))
+
+  implicit val formats = org.json4s.DefaultFormats ++ List(CustomDateTimeSerializer)
+  private val format = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss.SSSSSS")
+
+  /**
+   * Converts date string into Date object
+   */
+  private def stringToDate(dateString: String): DateTime = format.parseDateTime(dateString)
```

Comment:
I think my serializer wasn't right, I rewrote it and I'm now using this function inside it. Can you check if it looks ok?